x-pack/plugins/apm/public/services/rest/createCallApmApi.ts

```diff
@@ -28,16 +32,18 @@ export let callApmApi: APMClient = () => {
 
 export function createCallApmApi(http: HttpSetup) {
   callApmApi = ((options: APMClientOptions) => {
-    const { pathname, params = {}, ...opts } = options;
+    const { endpoint, params = {}, ...opts } = options;
 
     const path = (params.path || {}) as Record<string, any>;
+    const [method, pathname] = endpoint.split(' ');
```

Comment:
This will cause problems if the endpoint contains spaces. Something that's theoretically possible but generally frowned upon (for good reasons).
I don't think we need to handle this - just wanted to point it out so I can link to this conversation when this causes a production incident 5 years from now :D

Reply:
I have enough faith in humanity to assume that no one will ever try to register an endpoint with spaces 😅